src/main/java/org/prebid/server/bidder/adquery/AdQueryBidder.java

```diff
@@ -0,0 +1,213 @@
+                .bidder(BIDDER_NAME)
+                .bidderRequestId(bidRequest.getId())
+                .bidRequestsCount(1)
+                .bidderRequestsCount(1)
+                .sizes(getImpSizes(imp))
+                .bidIp(ObjectUtil.getIfNotNull(bidRequest.getDevice(), Device::getIp))
+                .bidIpv6(ObjectUtil.getIfNotNull(bidRequest.getDevice(), Device::getIpv6))
+                .bidUa(ObjectUtil.getIfNotNull(bidRequest.getDevice(), Device::getUa))
+                .bidPageUrl(ObjectUtil.getIfNotNull(bidRequest.getSite(), Site::getPage))
+                .build();
+
+    }
+
+    private String getImpSizes(Imp imp) {
+        final Banner banner = imp.getBanner();
+        if (banner == null) {
+            return StringUtils.EMPTY;
+        }
+
+        final List<Format> format = banner.getFormat();
+        if (CollectionUtils.isNotEmpty(format)) {
+            final List<String> sizes = new ArrayList<>();
+            format.forEach(singleFormat -> sizes.add(
+                    "%sx%s".formatted(getIntOrElseZero(singleFormat.getW()), getIntOrElseZero(singleFormat.getH()))));
+            return String.join("_", sizes);
+        }
+
+        final Integer w = banner.getW();
+        final Integer h = banner.getH();
+        if (w != null && h != null) {
+            return "%sx%s".formatted(w, h);
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private int getIntOrElseZero(Integer number) {
+        return number != null ? number : 0;
+    }
+
+    private MultiMap resolveHeader(Device device) {
+        final MultiMap headers = HttpUtil.headers();
+        headers.add(HttpUtil.X_OPENRTB_VERSION_HEADER, ORTB_VERSION);
+
+        if (Objects.nonNull(device) && StringUtils.isNotBlank(device.getIp()) && device.getIp().length() > 0) {
+            headers.add(HttpUtil.X_FORWARDED_FOR_HEADER, device.getIp());
+        }
+
+        return headers;
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(BidderCall<AdQueryRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final AdQueryResponse bidResponse = mapper.decodeValue(
+                    httpCall.getResponse().getBody(), AdQueryResponse.class);
+            return Result.withValues(extractBids(bidResponse, bidRequest));
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(AdQueryResponse adQueryResponse, BidRequest bidRequest) {
+        if (adQueryResponse == null || Objects.isNull(adQueryResponse.getData())) {
+            return Collections.emptyList();
+        }
+
+        final AdQueryDataResponse data = adQueryResponse.getData();
+        final Bid bid = Bid.builder()
+                .id(data.getRequestId())
+                .impid(resoleImpId(bidRequest, data))
+                .price(BigDecimalParser.parse(data.getCpm()))
```

Comment:
I believe we can expect BigDecimal on the DTOs side and let the jackson mapper do the job